### DISCUSSION TITLE
Trail Sense Nightly Builds

### DISCUSSION THREAD

**Discussion by kylecorry31** [kylecorry31/Trail-Sense#1940] [Announcements]
Trail Sense Nightly builds are generated every night off of the main branch. **These builds should only be used for testing purposes and they may contain bugs or unfinished features.** 

For reference, I've been exclusively running builds off of the main branch for over 3 years on my device - so I try to keep it pretty stable.

I'm still researching how to add a description of what the build contains compared to the latest release.

To try out a nightly build, visit https://github.com/kylecorry31/Trail-Sense-Nightly and download the latest APK from the releases page. You can also choose to download from https://github.com/kylecorry31/Trail-Sense/actions/workflows/nightly.yml and click the latest Nightly run. The APK will be attached as an artifact (bottom of the page), and you can download + extract that to get the APK file. When you install the APK file on your device, it will install along side other versions of Trail Sense, so you don't need to worry about it overwriting data.

Builds are signed by me using a new nightly keystore. Seems like if you have Play Protect enabled on your device, it may flag it as an unknown signature - if you click more info, there will be a not very visible button to install anyway.

Nightly builds are deleted after 7 days.

If you find an issue with the nightly build, please check the current milestone for the related issue and leave a comment there. If you can't find an appropriate issue, you can post here or create a new issue (just mention Nightly in the title). 

I'll occasionally post here if a large feature could use extra testing or feedback.

There's also a nightly release APK on the action which is the optimized release build off of main.

**Note:** This will install as a separate app with a modified icon so you can tell them apart.

**Note:** If you try to restore a backup made by a nightly build in an older release build it may not work. Avoid running nightly as your primary app version.

**Comment by kylecorry31** [comment]
If you wanted to tested the nightly out using the data in your normal Trail Sense install, just use the backup / restore feature (Settings > Backup / Restore). Backup your normal Trail Sense and then import that file using the restore feature of the nightly app.

**Comment by kylecorry31** [comment]
Dynamic Color support (Android 12+) will be available in tonight's nightly. It is off by default and there's a new Dynamic Colors setting toward the bottom of the settings page.

**Comment by surendrajat** [comment]
Thank you for creating and maintaing this app. It's very useful.

You can use https://nightly.link/kylecorry31/Trail-Sense/workflows/nightly/main to directly access the nightly build.

**Reply by kylecorry31** [reply]
Thank you, this is really helpful!

**Comment by kylecorry31** [comment]
A new "Trail-Sense-Nightly-Release.apk" artifact has been added to the nightly builds. This has a different package ID than the other nightly so can be installed along side it (or any of the other Trail Sense versions). 

The intent of this one is for users who want the latest version of Trail Sense but without the overhead of debugging tooling.

As with all nightly builds, they should not be relied on, though this will likely be the one I use personally so it should be in pretty good shape (I was using the debug build, but that's much slower and heavier on battery usage).
